src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -349,6 +353,8 @@ private async Task<IConfidentialClientApplication> BuildConfidentialClientApplic
 
             try
             {
+                CheckApplicationOptionsContainClientSecret(_applicationOptions);
```

Comment:
TLDR: What about certificates?



There are 4 ways of specifying the secure contract between the server and AAD in a confidential client: 

- client secret (should not really be used in production)
- cert (recommended) 
- client assertion (which is essentially a string with cert details - we use this when MSAL can't handle wierd certs)
- cert + extra claims 

ConfidentialClientApplicationOptions only allows users to pass in a secret. But you must provide mechanism for user to configure certificates.

Reply:
@bgavrilMS : yes, and we have an enhancements issue for this: (#12).

For the moment, proposing an incremental approach